_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+
+{% include toc.md %}
+
+## Understanding Containers vs. Virtual Machines
+
+Around 2013, [Docker](https://www.docker.com/) was released, beginning a shift in how we think about hosting applications and managing infrastructure. Infrastructure teams and operations have been leveraging Virtual Machines with great success for decades, so you may be tempted to think, "isn't this just a virtual machine?" In spirit, yes; in application, no.
+
+To talk about containers, we have to talk about Virtual Machines. A virtual machine (VM) is a virtual representation of a computer all the way from its boot process to loading an entire operating system. This provides a lot of flexibility because you gain fine-grained control over an entire virtual computer, but it brings all the pitfalls of a whole computer as well. The operating system has to be maintained, patched, and updated; disks have to be managed, and all manner of operational overhead goes into their care and feeding. Notably, VMs are also huge because an entire OS has to go on them. So when you start a VM, you have to go through the entire boot sequence of a traditional computer.
+
+These are the three problem areas that containers hope to solve; Ease of management, size, and speed to start. There are others, but for simplicity, we will focus on those three as I feel (IMHO) they are the most relevant to business processes. Containers differ from a traditional virtual machine in one significant way. They do not have an actual Operating System. This may come as a surprise to you, even if you might have dabbled with containers before. Containers instead have an abstraction of an operating system that provides hooks into the host operating system using standards and conventions supplied by the container system.
+
+Docker is the de facto container standard for all practical purposes, but other container technologies do exist. The notion of a standardized container comes from the shipping industry. In shipping, there was no standardization, which made transport difficult because you had to figure out how to hold each item on a boat, train, or truck. Containers provided a standard format for moving goods "you can put whatever you want inside the container so long as it conforms to a standard set of dimensions and can be opened and bolted down in a uniform way." This simplified the logistical process of moving real-world items. In this same way, standardizing the way a software container interacts with its host OS to delegate the responsibility for executing tasks simplifies management of the most crucial part, the application.
```

Comment:
```suggestion
Docker is the de facto container standard for all practical purposes, but other container technologies do exist. The notion of a standardized container comes from the shipping industry. In shipping, there was no standardization, which made transport difficult because you had to figure out how to hold each item on a boat, train, or truck. Containers provided a standard format for moving goods: "You can put whatever you want inside the container so long as it conforms to a standard set of dimensions and can be opened and bolted down in a uniform way." This simplified the logistical process of moving real-world items. In this same way, standardizing the way a software container interacts with its host OS to delegate the responsibility for executing tasks simplifies management of the most crucial part, the application.
```
punctuation